tests/positive/filesystems/skip_readonly.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filesystems
+
+import (
+	"github.com/coreos/ignition/v2/tests/register"
+	"github.com/coreos/ignition/v2/tests/types"
+)
+
+func init() {
+	register.Register(register.PositiveTest, SkipReadOnlyFilesystems())
+}
+
+func SkipReadOnlyFilesystems() types.Test {
+	name := "filesystem.skip.readonly"
+
+	mountPathName := "/usr/share/oem1"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	mntDevices := []types.MntDevice{
+		{
+			Label:        "OEM",
+			Substitution: "$DEVICE",
+		},
+	}
+	config := `{
+		"ignition": {"version": "$version"},
+		"storage": {
+			"filesystems": [{
+				"path": "/dev/sda1",
+				"device": "$DEVICE",
+				"format": "ext4",
+				"wipeFilesystem": true,
+				"label": "READ-ONLY"
+			  }]
+		}
+	}`
+
+	in[0].Partitions.GetPartition("READ-ONLY").FilesystemType = "ext4"
+	out[0].Partitions.GetPartition("READ-ONLY").FilesystemType = "ext4"
+	in[0].Partitions.GetPartition("READ-ONLY").FilesystemMode = 0444
```

Comment:
This just removes the write bit from the directory permissions, but Ignition runs as the root user, and root ignores permission bits.  A read-only filesystem is one mounted with `-o ro`.

And in any event, we need the filesystem containing the mountpoint to be read-only, not the one being mounted.

Reply:
Okay didn't know ignition ran as a root user,
I thought i could just assign the permission bits to a variable and check against them with the partition i made in types.go

This will be dropped in the next push since Blackbox tests aren't needed